### DISCUSSION TITLE
Feature Request: V1 - Add SSL Email Support

### DISCUSSION THREAD

**Discussion by nkringle** [mealie-recipes/mealie#1218] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Essentially my email server/setup uses the standard email protocol SSL/port 465. It looks like mealie supports SmartTLS and unauthenticated email, just not authenticated SSL SMTP. 

**(Optional): Suggest A Solution**
I'm not super proficient in python, but I think this would be relatively easy to implement:

1. According to [python email's docs](https://python-emails.readthedocs.io/en/latest/), you just need to add a `dict` value of `'ssl': True`. It looks like that could go somewhere around [here](https://github.com/hay-kot/mealie/blob/67178f9b7455972f7730a699018a72cf39190d18/mealie/services/email/email_senders.py#L27-L28).
2. That probably means that there should be a [new environment variable/config option](https://github.com/hay-kot/mealie/blob/67178f9b7455972f7730a699018a72cf39190d18/mealie/core/settings/settings.py#L68-L89) for `SMTP_SSL` and would need to be added to the constructor and also the docs & docker compose files. There should probably be logic that does not allow both `SMTP_TLS` and `SMTP_TLS` to both be true. 

I think there is also an opportunity for creating a better developer experience by changing the email authentication strategy to a single environment variable: `SMTP_AUTH_STRATEGY` (for example), that takes either `None` (default), `TLS`, or `SSL`. That may be a bigger change since that would break any current users. You could probably do something where if someone set TLS = true you use that to default SMTP_AUTH to TLS, but again that depends on how you want to transition support for this. 

**Additional Information**
- [x] If the feature is accepted I'm willing to submit a PR to provide this feature. 
-> Sure, I would want to know which of the above two strategies you want this to go. :)

**Comment by nkringle** [comment]
I think there is tech debt that could be picked up with this too, if authentication is 'none' then don't expect an SMTP user and password. You probably also could move the 'smtp_tls' variable out of the required check and just let it default to false.

**Comment by hay-kot** [comment]
I think 2 is a good option. I'm okay with the breaking change. I wonder if there's a way to fallback to the TLS setting if we don't detect an SMTP_AUTH_STRATEGY? I also agree that there's some other technical debt in there. 

Let me know if you need anything else to get started on this one.

**Comment by nkringle** [comment]
Hey @hay-kot, I created a PR ( #1235 ) for this, I left a couple preliminary questions in there.
